interop/ksp/src/main/kotlin/com/squareup/kotlinpoet/ksp/ksTypes.kt

```diff
@@ -53,7 +53,8 @@ public fun KSType.toClassName(): ClassName {
  */
 @KotlinPoetKspPreview
 public fun KSType.toTypeName(
-  typeParamResolver: TypeParameterResolver = TypeParameterResolver.EMPTY
+  typeParamResolver: TypeParameterResolver = TypeParameterResolver.EMPTY,
+  typeArguments: List<KSTypeArgument> = emptyList(),
```

Comment:
Not sure how to handle this api-wise. I just added it as an extra argument to get this info but there may be a better way.

Reply:
Let's keep the original public signature but then move the full impl with this overload to an internal separate function. We can open it up later if there's a need for it